pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -232,9 +232,11 @@ func (rc *reconciler) reconcile() {
 					metrics.RecordForcedDetachMetric()
 					klog.Warningf(attachedVolume.GenerateMsgDetailed("attacherDetacher.DetachVolume started", fmt.Sprintf("This volume is not safe to detach, but maxWaitForUnmountDuration %v expired, force detaching", rc.maxWaitForUnmountDuration)))
 				}
-			}
-			if err != nil && !exponentialbackoff.IsExponentialBackoff(err) {
-				// Ignore exponentialbackoff.IsExponentialBackoff errors, they are expected.
+			} else if exponentialbackoff.IsExponentialBackoff(err) {
+				// not log exponentialbackoff.IsExponentialBackoff errors, they are expected.
+				// add volume back to ReportAsAttached list
+				rc.actualStateOfWorld.AddVolumeToReportAsAttached(attachedVolume.VolumeName, attachedVolume.NodeName)
+			} else {
```

Comment:
how about here, if other errors, we should also add the volume back to the list

Reply:
> how about here, if other errors, we should also add the volume back to the list

I think you r right. But except AlreadyExistsError. AlreadyExistsError means  an operation with the same operation name is already executing, after this operation, controller will add the volume back to the list

Reply:
From the logic, I don't think AlreadyExsistError should happen, but I am ok for checking it here.